src/app/Card.tsx

```diff
@@ -1,33 +1,38 @@
 import { useEffect, useState } from 'react';
+import { FaTimes } from 'react-icons/fa';
+import { ansiFormat } from '../support/formatting';
+import { getWorkingDirectory } from '../support/nana';
 import { Output } from './Output';
 import { Prompt } from './Prompt';
-import { getWorkingDirectory } from '../support/nana';
-import { ansiFormat } from '../support/formatting';
-import { FaTimes } from 'react-icons/fa';
 
 export type CardPropTypes = {
   workingDir?: string;
-  input?: string;
+  input: string;
   output?: string;
 };
 
 export type ICard = CardPropTypes & {
-  id: number;
+  id: string;
 };
 
-export const Card = ({
-  input: initialInput,
-  workingDir,
-  history,
-  output,
-  onClose,
-  onSubmit,
-}: ICard & {
-  history: string[];
-  onSubmit: (newProps: Partial<CardPropTypes>, isError: boolean) => void;
-  onClose: () => void;
-}) => {
-  const [input, setInput] = useState(initialInput);
+export const Card = (
+  props: ICard & {
+    history: string[];
+    onInputChange: (newInput: string) => void;
+    onSubmit: (newProps: CardPropTypes, isError: boolean) => void;
+    onClose: () => void;
+  }
+) => {
```

Comment:
I find this easier to read than the old destructuring-inside-parens approach, thanks.